File: Frontend/src/components/Models/Purchase.ts
import type { Item } from "./Item";
import type { User } from "./User";

export interface Purchase {
    id: number;
    userId: number;
    user: User; // Asumiendo que ya tienes una interfaz IUser para el modelo User
    itemId: number;
    item: Item; // Asumiendo que ya tienes una interfaz IItem para el modelo Item
    purchaseDate: string; // Usamos string para representar la fecha en formato ISO
}

File: Backend/Model/Entities/Purchase.cs
namespace AA2_CS.Model.Entities
{
    public class Purchase
    {
        public int id { get; set; }
        public int userid { get; set; }
        public int itemid { get; set; }
        public Item? Item { get; set; }
        public User? User { get; set; }
        public DateTime purchasedate { get; set; } = DateTime.UtcNow;

        public Purchase(int userId, User user, int itemId, Item item, DateTime purchaseDate)
        {
            this.userid = userId;
            this.itemid = itemId;
            this.purchasedate = purchaseDate.Kind == DateTimeKind.Utc
                ? purchaseDate
                : purchaseDate.ToUniversalTime();
        }

        public Purchase() { }
    }
}
